docs/transform.md

```diff
@@ -0,0 +1,55 @@
+# Transforms
+
+## Summary
+
+Transforms map or filter the input or output of a child process. They are defined by passing an [async generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function*) to the [`stdin`](../readme.md#stdin), [`stdout`](../readme.md#stdout-1), [`stderr`](../readme.md#stderr-1) or [`stdio`](../readme.md#stdio-1) option.
+
+```js
+import {execa} from 'execa';
+
+const transform = async function * (chunks) {
+	for await (const chunk of chunks) {
+		yield chunk.toUpperCase();
+	}
+};
+
+const {stdout} = await execa('echo', ['hello'], {stdout: transform});
+console.log(stdout); // HELLO
+```
+
+## Encoding
+
+The `chunks` argument passed to the transform is an [`AsyncIterable<string>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_async_iterator_and_async_iterable_protocols). If the [`encoding`](../readme.md#encoding) option is `buffer`, it is an `AsyncIterable<Uint8Array>` instead.
+
+The transform can `yield` either a `string` or an `Uint8Array`, regardless of the `chunks` argument's type.
+
+## Filtering
+
```

Comment:
Users must be careful hereful here. Their secret could potentially be split over multiple chunks. This example would work better if the chunk was a line.

Reply:
Yes, absolutely, great catch!
I did think of it as well. I have an upcoming PR which will provide with line-wise splitting for the generators `chunks`.